weave/trace_server/interface/base_object_classes/annotation_spec.py

```diff
@@ -41,12 +42,49 @@ class AnnotationSpec(base_object_def.BaseObject):
         ],
     )
 
-    @field_validator("json_schema")
-    def validate_json_schema(cls, v: dict) -> dict:
+    @model_validator(mode="before")
+    @classmethod
+    def preprocess_field_schema(cls, data: dict[str, Any]) -> dict[str, Any]:
+        if "field_schema" not in data:
+            return data
+
+        field_schema = data["field_schema"]
+
+        # Handle Pydantic Field
+        if isinstance(field_schema, FieldInfo):
+            data["field_schema"] = field_schema.json_schema()
+            return data
+
+        # Handle Pydantic model
+        if isinstance(field_schema, type) and issubclass(field_schema, BaseModel):
+            data["field_schema"] = field_schema.model_json_schema()  # type: ignore
```

Comment:
We currently don't support nested objects, we might want to throw with an actionable error mentioning they will have to create multiple specs. I will add a ticket to move as much of the human feedback UI to the Zod form, which will allow for nested.

Reply:
I think we should allow the creation, but maybe in the UI there is a fallback that allows free-form json input or something